debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorIT.java

```diff
@@ -1054,6 +1058,57 @@ public void shouldIgnoreCreateIndexForNonCapturedTablesNotStoredInHistory() thro
         Assertions.assertThat(record.topic()).isEqualTo(DATABASE.topicForTable("customers"));
     }
 
+    @Test
+    @FixFor("DBZ-683")
+    public void shouldReceiveSchemaForNonWhitelistedTablesAndDatabases() throws SQLException, InterruptedException {
+        Testing.Files.delete(DB_HISTORY_PATH);
+
+        final String tables = String.format("%s.customers,%s.orders", DATABASE.getDatabaseName(), DATABASE.getDatabaseName());
+        config = DATABASE.defaultConfig()
+                              .with(MySqlConnectorConfig.TABLE_WHITELIST, tables)
+                              .with(MySqlConnectorConfig.SNAPSHOT_MODE, SnapshotMode.SCHEMA_ONLY)
+                              .with(MySqlConnectorConfig.INCLUDE_SCHEMA_CHANGES, true)
+                              .with(MySqlConnectorConfig.DATABASE_WHITELIST, ".*")
+                              .build();
+
+        dropDatabases();
+
+        try (MySQLConnection db = MySQLConnection.forTestDatabase("mysql");) {
+            try (JdbcConnection connection = db.connect()) {
+                connection.execute(
+                        "CREATE DATABASE non_wh",
+                        "USE non_wh",
+                        "CREATE TABLE t1 (ID INT PRIMARY KEY)"
+                );
+            }
+        }
+
+        // Start the connector ...
+        start(MySqlConnector.class, config);
+
+        // Consume the first records due to startup and initialization of the database ...
+        // Testing.Print.enable();
+        // Four tables in database, only two are whitelisted
+        SourceRecords records = consumeRecordsByTopic(1 + 3 + 2 * 4 + 3 + 2);
```

Comment:
So I think these additions help a *little bit*, but still it's not quite clear what the expected value represents. Can we clarify this in terms of docs somehow? I don't have a good answer, but the current form is really hard to grasp.

Reply:
@gunnarmorling Updated